Sources/Rendering/OpenGL/ShaderProgram/index.js

```diff
@@ -572,14 +577,9 @@ function extend(publicAPI, model, initialValues = {}) {
   Object.assign(initialValues, defaultValues(initialValues));
 
   // Instantiate internal objects
-  model.attributesLocs = {};
-  model.uniformLocs = {};
-  model.vertexShader = vtkShader.newInstance();
-  model.vertexShader.setShaderType('Vertex');
-  model.fragmentShader = vtkShader.newInstance();
-  model.fragmentShader.setShaderType('Fragment');
-  model.geometryShader = vtkShader.newInstance();
-  model.geometryShader.setShaderType('Geometry');
+  initialValues.vertexShader.setShaderType('Vertex');
```

Comment:
can't shader type be passed as initialValue ?

Reply:
It was not the case before, but as now the shaders could be passed in initialValues it would be logical to accept setting the shaderType.